README.md

```diff
@@ -129,17 +108,77 @@ spec:
   * For per port parameters (suffix `_P1`, `_P2`) parameters with `_P2` suffix are ignored if the device is single port.
 * If a configuration is not set in spec, its non-volatile configuration parameters (if any) should be set to device default.
 
+### NicFirmwareSource
+
+The [NICFirmwareSource CR](/api/v1alpha1/nicfirmwaresource_types.go) represents a list of url sources with NIC FW binaries archives.
+
+#### Provisioning a storage class for NIC FW upgrade
+
+To enable the NIC FW upgrade feature, `NicFirmwareStorage` section should be provided in the [NIC Cluster Policy](https://github.com/Mellanox/network-operator/blob/7c31e789835afdad114b1a68011e5307bc773bcf/api/v1alpha1/nicclusterpolicy_types.go#L321). There is an option to create a new PVC or use the existing one.
+Firmware binaries will be provisioned by a provisioner controller which will watch for NICFirmwareSource obj and provision the binaries in a shared volume enabled by the given storage class.
+Node agents will need to make sure that the reference NICFirmwareSource object is fully reconciled (status.state == Success) before proceeding with firmware update.
+
+#### Example of the storage class deployment
+
+To set up a persistent NFS storage in the cluster, the [example from the CSI NFS Driver repository](https://github.com/kubernetes-csi/csi-driver-nfs/blob/master/deploy/example/nfs-provisioner/README.md) might be used.
+After deploying the NFS server and NFS CSI driver, the [storage class](https://github.com/kubernetes-csi/csi-driver-nfs/blob/master/deploy/example/storageclass-nfs.yaml) should be deployed. The name of the storage class can then be passed to the NIC Configuration Operator helm chart.
```

Comment:
Should we still mention Helm chart or should it be NicClusterPolicy?

Reply:
Thanks, changed to NIC Cluster Policy